OEP-6/OEP-6.mediawiki

```diff
@@ -0,0 +1,452 @@
+
+====getDefaultIdentity====
+<pre>
+function getDefaultIdentity(): Promise<string>
+</pre>
+
+Returns currently selected identity of logged in user.
+
+* Rejects with '''NO_IDENTITY''' in case the user is not signed in or has no identity
+
+====getPublicKeys====
+<pre>
+function getPublicKeys(identity: string): Promise<string[]>
+</pre>
+Returns public keys corresponding to an <code>identity</code>. The <code>identity</code> '''does not''' need to be one of the identities returned by <code>getOwnIdentities</code>. This call must query blockchain by contrast to <code>asset.getPublicKey</code>.
+
+* Rejects with '''MALFORMED_IDENTITY''' in case the <code>identity</code> is not a valid identity
+
+====getDDO====
+<pre>
+interface OntIdDDO {
+    ...
+}
+
+function getDDO(identity: string): Promise<OntIdDDO>
+</pre>
+
+Queries Description Object of the <code>identity</code>. The <code>identity</code> '''does not''' need to be one of the identities returned by <code>getOwnIdentities</code>. This operation is not signed and therefore does not require user interaction.
+
+* Rejects with '''MALFORMED_IDENTITY''' in case the <code>identity</code> is not a valid identity
+
+====getAttributes====
+<pre>
+interface OntIdAttribute {
+    key: string;
+    type: string;
+    value: string;
+}
+
+function getAttributes(identity: string): Promise<OntIdAttribute[]>
+</pre>
+
+Queries attributes attached to the <code>identity</code>. The <code>identity</code> '''does not''' need to be one of the identities returned by <code>getOwnIdentities</code>. This operation is not signed and therefore does not require user interaction.
+
+* Rejects with '''MALFORMED_IDENTITY''' in case the <code>identity</code> is not a valid identity
+
+====addAttributes====
+<pre>
+function addAttributes(identity: string, attributes: OntIdAttribute[]): Promise<void>
+</pre>
+
+Adds attributes to the <code>identity</code>. The <code>identity</code> must be one of the identities returned by <code>getOwnIdentities</code>.
+
+* Rejects with '''NO_IDENTITY''' in case the user is not signed in or has no identity
+* Rejects with '''WRONG_IDENTITY''' in case the <code>identity</code> is not one of own identites
+* Rejects with '''MALFORMED_IDENTITY''' in case the <code>identity</code> is not a valid identity
+
+====removeAttributes====
+<pre>
+function removeAttribute(identity: string, key: string): Promise<void>
+</pre>
+
+Removes attribute from the <code>identity</code>. The <code>identity</code> must be one of the identities returned by <code>getOwnIdentities</code>.
+
+* Rejects with '''NO_IDENTITY''' in case the user is not signed in or has no identity
+* Rejects with '''WRONG_IDENTITY''' in case the <code>identity</code> is not one of own identites
+* Rejects with '''MALFORMED_IDENTITY''' in case the <code>identity</code> is not a valid identity
+
+===Message===
+This API deals with arbitrary message signing and verification.
```

Comment:
Actually this is not about raw transaction. Signing raw transaction is very dangerous and should not be allowed at all. 

This is about encrypted messaging. There is nothing in the Ontology documentation, but it is modelled after Ethereum message signing.

Reply:
Thanks for the clarification on this. After reviewing more documentation of related protocols for secure message signing techniques, I think we should remove this from OEP-6, and create a new OEP specifically relating to message signing. Reason being is that message signing is can expand beyond the scope of dAPI providers, and there may be others looking to implement message signing. Because of that, it would be good to have a standard set for the different types of message signing formats that will be present in the Ontology ecosystem.

At some later point, once the OEP for message signing is accepted, we can file a new OEP to add the message signing functions to the dAPI standards.